packages/core/src/domain/session/sessionCookieStore.spec.ts

```diff
@@ -225,5 +225,31 @@ describe('session cookie store', () => {
         done()
       }, maxNumberOfRetriesShouldBeReached)
     })
+
+    it('should execute cookie accesses in order', (done) => {
+      lockScenario({
+        onInitialLockCheck: () => ({
+          currentState: { ...initialSession, lock: 'locked' }, // force to postpone first access
+          retryState: initialSession,
+        }),
+      })
+      persistSession(initialSession, COOKIE_OPTIONS)
+
+      withCookieLockAccess({
+        options: COOKIE_OPTIONS,
+        process: (session) => ({ ...session, first: 'added' }),
+        after: afterSpy,
+      })
+      withCookieLockAccess({
+        options: COOKIE_OPTIONS,
+        process: (session) => ({ ...session, second: 'added' }),
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏maybe write the cookie based on the previous value, ex:

```
withCookieLockAccess({
   ...
 process: (session) => ({ ...session, value: 'foo' }),
 ...
})
withCookieLockAccess({
 ...
 process: (session) => ({ ...session, value: session.value + 'bar' }),
 ...
// then check if value is `foobar`
```

So we know that both actions have been called in the good order.

Reply:
indeed 👍